test/conformance/tests/test_policy_configuration.go

```diff
@@ -0,0 +1,790 @@
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/pod"
+)
+
+var _ = Describe("[sriov] operator", Ordered, func() {
+	var sriovInfos *cluster.EnabledNodes
+
+	BeforeAll(func() {
+		isSingleNode, err := cluster.IsSingleNode(clients)
+		Expect(err).ToNot(HaveOccurred())
+		if isSingleNode {
+			disableDrainState, err := cluster.GetNodeDrainState(clients, operatorNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			if discovery.Enabled() {
+				if !disableDrainState {
+					Skip("SriovOperatorConfig DisableDrain property must be enabled in a single node environment")
+				}
+			}
+			snoTimeoutMultiplier = 1
+			if !disableDrainState {
+				err = cluster.SetDisableNodeDrainState(clients, operatorNamespace, true)
+				Expect(err).ToNot(HaveOccurred())
+				clean.RestoreNodeDrainState = true
+			}
+		}
+
+		err = namespaces.Create(namespaces.Test, clients)
+		Expect(err).ToNot(HaveOccurred())
+		err = namespaces.Clean(operatorNamespace, namespaces.Test, clients, discovery.Enabled())
+		Expect(err).ToNot(HaveOccurred())
+		WaitForSRIOVStable()
+		sriovInfos, err = cluster.DiscoverSriov(clients, operatorNamespace)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Describe("Custom SriovNetworkNodePolicy", func() {
+		BeforeEach(func() {
+			err := namespaces.Clean(operatorNamespace, namespaces.Test, clients, discovery.Enabled())
+			Expect(err).ToNot(HaveOccurred())
+			WaitForSRIOVStable()
+		})
+
+		Describe("Configuration", func() {
+			Context("Create vfio-pci node policy", func() {
+				var vfioNode string
+				var vfioNic sriovv1.InterfaceExt
+
+				BeforeEach(func() {
+					if discovery.Enabled() {
+						Skip("Test unsuitable to be run in discovery mode")
+					}
+
+					vfioNode, vfioNic = sriovInfos.FindOneVfioSriovDevice()
+					if vfioNode == "" {
+						Skip("skip test as no vfio-pci capable PF was found")
+					}
+					By("Using device " + vfioNic.Name + " on node " + vfioNode)
+				})
+
+				It("Should be possible to create a vfio-pci resource and allocate to a pod", func() {
+					By("creating a vfio-pci node policy")
+					resourceName := "testvfio"
+					_, err := network.CreateSriovPolicy(clients, "test-policy-", operatorNamespace, vfioNic.Name, vfioNode, 5, resourceName, "vfio-pci")
+					Expect(err).ToNot(HaveOccurred())
+
+					By("waiting for the node state to be updated")
+					Eventually(func() sriovv1.Interfaces {
+						nodeState, err := clients.SriovNetworkNodeStates(operatorNamespace).Get(context.Background(), vfioNode, metav1.GetOptions{})
+						Expect(err).ToNot(HaveOccurred())
+						return nodeState.Spec.Interfaces
+					}, 1*time.Minute, 1*time.Second).Should(ContainElement(MatchFields(
+						IgnoreExtras,
+						Fields{
+							"Name":   Equal(vfioNic.Name),
+							"NumVfs": Equal(5),
+						})))
+
+					By("waiting the sriov to be stable on the node")
+					WaitForSRIOVStable()
+
+					By("waiting for the resources to be available")
+					Eventually(func() int64 {
```

Comment:
not specially to this PR but can you create a function for this we use it in multiple places already?

Reply:
Refactored to a common `waitForPodRunning(...)`. Got rid of a lot of code